akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2AlpnSupport.scala

```diff
@@ -42,7 +46,7 @@ private[http] object Http2AlpnSupport {
     engine.asInstanceOf[JDK9SSLEngine].setHandshakeApplicationProtocolSelector(new BiFunction[SSLEngine, ju.List[String], String] {
       // explicit style needed here as automatic SAM-support doesn't seem to work out with Scala 2.11
       override def apply(engine: SSLEngine, protocols: ju.List[String]): String = {
-        val chosen = chooseProtocol(protocols)
+        val chosen = chooseProtocol(protocols).orElse(null)
```

Comment:
Maybe add a comment that returning `null` here means aborting the handshake according to the [`setHandshakeApplicationProtocolSelector` javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/SSLEngine.html#setHandshakeApplicationProtocolSelector(java.util.function.BiFunction))

> If the return value is null (no value chosen) or is a value that was not advertised by the peer, the underlying protocol will determine what action to take. (For example, ALPN will send a "no_application_protocol" alert and terminate the connection.)